emcc.py

```diff
@@ -554,6 +554,8 @@ def get_binaryen_passes():
     passes += ['--fpcast-emu']
   if settings.ASYNCIFY:
     passes += ['--asyncify']
+    if settings.MAIN_MODULE or settings.SIDE_MODULE:
+      passes += ['--pass-arg=asyncify-side-module']
```

Comment:
I had not realized that this needs to be run on the main and the side modules, but I guess that makes sense... if so, should we call the param on the binaryen side `asyncify-relocatable`? Otherwise it seems odd to pass it for a main module here.

Reply:
Originally I've designed sharing globals with asyncify as that these globals are exported from a main module and are imported into side modules. When creating this patch in Emscripten side, I've found that it will be easier that making those globals are imported into both main and side modules, and that this implementation will be more consistent with `__stack_pointer`, `__memory_base` and `__table_base`, which are all imported into main/side modules with the option `-sRELOCATABLE`.

Reply:
So, I agree with renaming `asyncify-side-module` to `asyncify-relocatable` on the binaryen side.

Reply:
I've created new pull request of this renaming option. <https://github.com/WebAssembly/binaryen/pull/4596>